### PR TITLE
Use the dual phase number to define strain markers

### PR DESCRIPTION
**Pull request by tduretz**
Here's an example using the setup `Shrinking`.
Before:
image

After:
image

See relevant code in `Shrinking.c`:

```C
int SetDualPhase(MdoodzInput *input, Coordinates coordinate, int phase) {
    
    int    dual_phase = phase;
    double Lx = input->model.xmax - input->model.xmin;
    double Lz = input->model.zmax - input->model.zmin;
    double Ax, Az;

    // Set checkerboard for phase 0
    Ax = cos( 6.0*2.0*M_PI*coordinate.x / Lx  );
    Az = sin( 6.0*2.0*M_PI*coordinate.z / Lz  );
    if ( ( (Az<0.0 && Ax<0.0) || (Az>0.0 && Ax>0.0) ) && dual_phase==0 ) {
        dual_phase += input->model.Nb_phases;
    }

    // Set checkerboard for phase 1
    Ax = cos( 24.0*2.0*M_PI*coordinate.x / Lx  );
    Az = sin( 24.0*2.0*M_PI*coordinate.z / Lz  );
    if ( ( (Az<0.0 && Ax<0.0) || (Az>0.0 && Ax>0.0) ) && dual_phase==1 ) {
        dual_phase += input->model.Nb_phases;
    }

  return dual_phase;
}
```

### CODE REVIEW

**Comment by kulakovri** [comment]
I think It would be better to remove breakpoints, output and executable and keep in MDLIB only those files that are for the library

**Comment by tduretz** [comment]
thanks @kulakovri, did not realise this. I've cleaned the files. This branch is almost complete, just need to add your windows stuff in there. It looks safer to make it the new main branch, the current main is somewhat incomplete.

**Review by kulakovri — APPROVED** [review]

File: MDLIB/include/mdoodz.h [hunk 1]
```diff
@@ -138,27 +140,29 @@ typedef struct {
...
-typedef int (*SetDual_f)(MdoodzInput *input, Coordinates coordinates, int phase);
+typedef int    (*SetPhase_f)(MdoodzInput *input, Coordinates coordinates);
+typedef int    (*SetDualPhase_f)(MdoodzInput *input, Coordinates coordinates, int phase);
 typedef double (*SetGrainSize_f)(MdoodzInput *input, Coordinates coordinates, int phase);
 typedef double (*SetPorosity_f)(MdoodzInput *input, Coordinates coordinates, int phase);
 typedef double (*SetDensity_f)(MdoodzInput *input, Coordinates coordinates, int phase);
 typedef double (*SetXComponent_f)(MdoodzInput *input, Coordinates coordinates, int phase);
 typedef double (*SetPressure_f)(MdoodzInput *input, Coordinates coordinates, int phase);
 typedef double (*SetNoise_f)(MdoodzInput *input, Coordinates coordinates, int phase);
+typedef void   (*SetDefGrad_f)(double*, double*, double*, double*, MdoodzInput *input, Coordinates coordinates, int phase);
```
Comment (kulakovri) [inline MDLIB/include/mdoodz.h]: Only thing is it would be nice to add names to new parameters, put those 4 arguments after the phase as a tensor argument

**Review by kulakovri — CHANGES_REQUESTED** [review]

File: MDLIB/include/mdoodz.h (hunk 1 above)
Comment (kulakovri) [reply MDLIB/include/mdoodz.h]: What does SetDefGrad stands for?

Comment (tduretz) [reply MDLIB/include/mdoodz.h]: Very good suggestion! Now Tensor2D type is public and used for input function. 
DefGrad stands for Deformation Gradient. I think it's a sufficiently clear name. Anyone who wants to deal with finite strain computations can pick it up easily (especially once we will build up docs).

**Review by kulakovri — APPROVED** [review]

**Comment by tduretz** [comment]
No need to merge

**Pull request closed without merging**